pkg/frontend/mysql_cmd_executor.go

```diff
@@ -3285,15 +3286,21 @@ func (ses *Session) getSqlType(sql string) {
 	}
 }
 
-func (mce *MysqlCmdExecutor) processLoadLocal(ctx context.Context, param *tree.ExternParam, writer *io.PipeWriter) (err error) {
+func (mce *MysqlCmdExecutor) processLoadLocal(ctx context.Context, param *tree.ExternParam, writer *io.PipeWriter, earlyStop *atomic.Bool) (err error) {
+	ses := mce.GetSession()
+	proto := ses.GetMysqlProtocol()
 	defer func() {
 		err2 := writer.Close()
 		if err == nil {
 			err = err2
 		}
+		if earlyStop.Load() { // if true, means mo get inner error, so we need to close the connection to stop data transfer
+			err2 = proto.GetTcpConnection().Close()
```

Comment:
In the offline discuession, I made a mistake. You can not close the tcp connection directly here. Because the connection is shared by the Load statement and the data. So, I recommends to report a suitable error refering to the mysql's document.